src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,116 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import java.util.Comparator;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+        class ComparatorPersonsByAge implements Comparator<Person> {
+
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(), o2.getAge());
+            }
+        }
+
+        Arrays.sort(persons, new ComparatorPersonsByAge());
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(), o2.getAge());
+            }
+        });
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                if (o1.getLastName().equals(o2.getLastName())) {
+                    return o1.getFirstName().compareTo(o2.getFirstName());
+                }
+                return o1.getLastName().compareTo(o2.getLastName());
+            }
+        });
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        class PredicatePerson implements Predicate<Person>{
+            @Override
+            public boolean apply(Person person) {
+                return person.getAge() == 30;
+            }
+        }
+        Person person = FluentIterable.from(persons).firstMatch(new PredicatePerson()).get();
```

Comment:
If value is not present, the exception will be thrown. It's better to use ofNull() method.

Reply:
fixed